libs/core/async_cuda/include/hpx/async_cuda/cuda_event.hpp

```diff
@@ -19,66 +21,95 @@ namespace hpx { namespace cuda { namespace experimental {
     // of them at startup.
     struct cuda_event_pool
     {
-        static constexpr int initial_events_in_pool = 128;
+        static constexpr std::size_t initial_events_in_pool = 128;
 
         static cuda_event_pool& get_event_pool()
         {
             static cuda_event_pool event_pool_;
             return event_pool_;
         }
```

Comment:
May I suggest that you move the (`HPX_CORE_EXPORT`ed) implementation of this function into a source file? The rationale is that otherwise on some platforms (Mac, Windows) each executable module (shared library, executable) that will contain code calling this function will have its own instance of the `event_pool_` variable.